cryptography/src/bls12381/primitives/group.rs

```diff
@@ -284,21 +274,23 @@ impl Element for Scalar {
             blst_fr_mul(&mut self.0, &self.0, &rhs.0);
         }
     }
+}
 
-    fn serialize(&self) -> Vec<u8> {
-        let mut bytes = [0u8; SCALAR_LENGTH];
+impl Write for Scalar {
+    fn write(&self, buf: &mut impl BufMut) {
+        let mut bytes: [u8; Self::SIZE];
```

Comment:
You need to keep this as `let mut bytes = [0u8; Self::SIZE];`

Reply:
(it is filled in with the inner logic of `blst` in the unsafe calls)

Reply:
Not sure if there is a safe way to write directly to `BufMut` with the interface exposed by `blst`.